pkg/userop/client.go

```diff
@@ -376,96 +372,67 @@ func (c *backend) SendUserOp(ctx context.Context, op UserOperation) (<-chan Rece
 	return done, nil
 }
 
-// waitForUserOpEvent waits for a user operation to be committed on block.
-func waitForUserOpEvent(
+// subscribeUserOpEvent waits for a user operation to be committed on block.
+func subscribeUserOpEvent(
 	ctx context.Context,
-	pollPeriod time.Duration,
 	cancel context.CancelFunc,
 	client EthBackend,
 	done chan<- Receipt,
 	entryPoint common.Address,
 	userOpHash common.Hash,
 ) {
-	ticker := time.NewTicker(pollPeriod * time.Millisecond)
-	defer ticker.Stop()
-	defer close(done)
 
-	fromBlock, err := client.BlockNumber(ctx)
+	entryPointContract, err := entry_point_v0_6_0.NewEntryPoint(entryPoint, client)
 	if err != nil {
-		slog.Error("failed to get block number", "error", err)
+		slog.Error("failed to connect to the entry point contract", "error", err)
 		cancel()
-		return
 	}
 
-	query := ethereum.FilterQuery{
-		Addresses: []common.Address{entryPoint},
-		Topics:    [][]common.Hash{{}, {userOpHash}},
-		FromBlock: big.NewInt(int64(fromBlock)),
-	}
-
-	for {
-		select {
-		case <-ctx.Done():
-			slog.Error("timeout waiting for user operation event", "hash", userOpHash.Hex())
-			return
-		case <-ticker.C:
-			logs, err := client.FilterLogs(ctx, query)
-			if err != nil {
-				slog.Error("failed to filter logs", "error", err)
-				continue
-			}
+	eventCh := make(chan *entry_point_v0_6_0.EntryPointUserOperationEvent)
 
-			receipt, err := processLogs(logs, userOpHash)
-			if err != nil {
-				slog.Error("failed to process logs", "error", err)
-				return
-			}
-			if receipt != nil {
-				done <- *receipt
-				return
-			}
-		}
+	sub, err := entryPointContract.WatchUserOperationEvent(nil, eventCh, [][32]byte{[32]byte(userOpHash.Bytes())}, nil, nil)
```

Comment:
I think you can replace `[32]byte(userOpHash.Bytes())` with just `userOpHash`

Reply:
1b82e85770b5a4165caac68d265d83a2b58a8702